fabtests/ubertest/uber.c

```diff
@@ -330,298 +329,125 @@ static int ft_recv_test_info(void)
-		info->fabric_attr->prov_name));
-
-	ret = ft_exchange_uint32(skip, &remote_skip);
-	if (ret)
-		return ret;
-
-	return skip || remote_skip;
-}
-
-static int ft_transfer_subindex(int subindex, int *remote_idx)
+static int ft_recv_result(struct fi_info *info)
 {
-	int ret;
-
-	ret = ft_sock_send(sock, &subindex, sizeof subindex);
-	if (ret) {
-		FT_PRINTERR("ft_sock_send", ret);
-		return ret;
-	}
-
-	ret = ft_sock_recv(sock, remote_idx, sizeof *remote_idx);
+	int ret, err = 0;
+	ret = ft_sock_recv(sock, &err, sizeof err);
 	if (ret) {
 		FT_PRINTERR("ft_sock_recv", ret);
 		return ret;
 	}
+	if (err) {
+		printf("Ending test %d, result: %s\n", test_info.test_index,
+			fi_strerror(-err));
+	}
 
-	return 0;
+	return err;
 }
 
-static int ft_fw_process_list_server(struct fi_info *hints, struct fi_info *info)
+static int ft_server_setup(struct fi_info *hints, struct fi_info *info)
 {
-	int ret, subindex, remote_idx = 0, result = -FI_ENODATA, end_test = 0;
-	int server_ready = 0;
-	struct fi_info *open_res_info;
+	int ret = 0;
 
-	ret = ft_sock_send(sock, &test_info, sizeof test_info);
-	if (ret) {
-		FT_PRINTERR("ft_sock_send", ret);
-		return ret;
+	hints = fi_allocinfo();
+	if (!hints) {
+		ret = -FI_ENOMEM
+		goto err;
 	}
 
-	for (subindex = 1, fabric_info = info; fabric_info;
-	     fabric_info = fabric_info->next, subindex++) {
-
-		ret = ft_check_info(hints, fabric_info);
-		if (ret)
-			return ret;
-
-		/* Stores the fabric_info into a tmp variable, resolves an issue caused
-		*  by ft_accept with FI_EP_MSG which overwrites the fabric_info.
-		*/
-		open_res_info = fabric_info;
-		while (1) {
-			fabric_info = open_res_info;
-			ret = ft_open_res();
-			if (ret) {
-				FT_PRINTERR("ft_open_res", ret);
-				return ret;
-			}
+	ft_fw_convert_info(hints, &test_info);
 
-			if (!server_ready) {
-				server_ready = 1;
-				ret = ft_sock_send(sock, &server_ready, sizeof server_ready);
-				if (ret) {
-					FT_PRINTERR("ft_sock_send", ret);
-					return ret;
-				}
-			}
+	ret = fi_getinfo(FT_FIVERSION, ft_strptr(test_info.node),
+			 ft_strptr(test_info.service), FI_SOURCE, hints, &info);
+	if (ret) {
+		FT_PRINTERR("fi_getinfo", ret);
+		goto err;
+	}
 
-			ret = ft_sock_recv(sock, &end_test, sizeof end_test);
-			if (ret) {
-				FT_PRINTERR("ft_sock_recv", ret);
-				return ret;
-			}
-			if (end_test) {
-				ft_cleanup();
-				break;
-			}
+	fabric_info = info;
 
-			if (ft_skip_info(hints, fabric_info)) {
-				ft_cleanup();
-				continue;
-			}
+	ret = ft_check_info(hints, fabric_info);
+	if (ret) 
+		goto err;
 
-			ret = ft_transfer_subindex(subindex, &remote_idx);
-			if (ret)
-				return ret;
+	ret = ft_open_res();
+	if (ret) 
+		goto err;
 
-			ft_fw_update_info(&test_info, fabric_info, subindex);
+	ft_fw_update_info(&test_info, fabric_info);
 
-			printf("Starting test %d-%d-%d: ", test_info.test_index,
-				subindex, remote_idx);
-			ft_show_test_info();
+	return 0;
+	err:
+		ft_send_result(ret, info);
+		return ret;
```

Comment:
nit: outdent everything from err: down.  Also, there's an extra space above after the if (ret) statement about 8 lines up.